test/e2e/tests/conftest.py

```diff
@@ -0,0 +1,148 @@
+
+
+def wait_container_running(container: docker.models.containers.Container, additional_wait):
+    retries = 6
+    container.reload()
+
+    while container.status != 'running':
+        retries -= 1
+        if retries == 0:
+            raise TimeoutError
+
+        sleep(0.5)
+        container.reload()
+
+    if additional_wait:
+        sleep(additional_wait)
+
+
+@pytest.fixture(scope="function")
+def run_containers(request, docker_client):
+    """
+    Runs containers, dumps their logs and cleans'em up
+    """
+    containers = {}
+    post = {}
+
+    for name, container in request.param.items():
+        image = container['image']
+        args = container.get('args', '')
+        privileged = container.get('privileged', False)
+        mounts = container.get('mounts', [])
+        environment = container.get('env', {})
+        additional_wait = container.get('init_wait', 0)
+        post_validation = container.get('post_validation', None)
+        stop_signal = container.get('signal', None)
+
+        handle = docker_client.containers.run(
+            image,
+            args,
+            name=name,
+            detach=True,
+            privileged=privileged,
+            mounts=mounts,
+            environment=environment
+        )
+
+        containers[name] = handle
+        post[name] = {
+            'validation': post_validation,
+            'signal': stop_signal
+        }
+
+        wait_container_running(handle, additional_wait)
+
+    yield containers
+
+    success = True
+    errors = []
+
+    for name, container in containers.items():
+        validation = post[name]['validation']
+        stop_signal = post[name]['signal']
+
+        if stop_signal:
+            container.kill(stop_signal)
+
+        # The stop command is issued regardless of the kill command to ensure
+        # the container stops
+        container.stop()
+
+        logs = container.logs().decode('ascii')
+        if logs:
+            with open(os.path.join(LOGS_PATH, f'{name}.log'), 'w') as f:
+                f.write(logs)
+
+        if validation:
+            res, msg = validation(container)
+            if not res:
+                errors.append(f'{name}: {msg}')
+                success = False
+
+        container.remove()
+
+    assert success, '\n'.join(errors)
+
+
+def pytest_html_report_title(report):
+    report.title = "sinsp e2e tests"
+
+
+def dump_logs(pytest_html, extra):
+    """
+    Finds all logs dumped to LOGS_PATH and makes them available through the
+    auto-generated report
+    """
+    for file in os.listdir(LOGS_PATH):
+        full_path = os.path.join(LOGS_PATH, file)
+        if not os.path.isfile(full_path):
+            continue
+
+        with open(full_path, 'r', errors='replace') as f:
+            logs = f.read()
+            extra.append(pytest_html.extras.text(logs, name=file))
+
+        # Remove file so it doesn't bleed to following tests
+        os.remove(full_path)
```

Comment:
I think this is highly unlikely.

`full_path` is formed by concatenating `LOGS_PATH` and `file`, `file` is acquired by listing `LOGS_PATH` and there is a check ensuring `full_path` is an actual file, so the `open` operation should pretty much succeed always, unless some other process in the tester container removed it at just the right time and that to me sounds like a situation I want a hard exception to occur (cause it is either a design error, race condition or some other thing I want to be fully aware of).

`os.remove` on the other hand, if we are going to run in something like Windows then we might want to `try/except` it in case something else opened the file (Windows hates people deleting opened files). But in unix systems I think the only possible error would be that the file (which we just read) doesn't exist and I think I would like it to be extremely explicit as well in this situation.

WDYT?